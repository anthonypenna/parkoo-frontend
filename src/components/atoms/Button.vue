<template>
  <button v-bind="$attrs" :class="`button button--${theme}`">
    <slot />
  </button>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'Button',

  props: {
    theme: {
      default: 'secondary',
      type: String
    }
  }
})
</script>

<style lang="scss" scoped>
.button {
  max-width: 295px;
  width: 100%;
  padding: 14px 0;
  border: none;
  border-radius: 5px;
  font-family: inherit;
  font-size: 16px;
  font-weight: bold;

  &--primary {
    background: #ff9979;
    color: #fff;
  }

  &--secondary {
    background: #f1f1f1;
    color: #000;
  }
}
</style>
