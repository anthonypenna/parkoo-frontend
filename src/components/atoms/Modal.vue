<template>
  <Fade>
    <div class="modal">
      <slot />
    </div>
  </Fade>
</template>

<script lang="ts">
import { Map } from "mapbox-gl";
import Vue from "vue";
import Fade from "@/components/atoms/Fade.vue";
import { mapGetters } from "vuex";

export default Vue.extend({
  name: "Modal",

  components: { Fade },

  computed: {
    ...mapGetters("mapbox", ["map"]),
  },

  mounted() {
    const map = this.map as Map;
    map.dragPan.disable();
    map.scrollZoom.disable();
  },

  destroyed() {
    const map = this.map as Map;
    map.dragPan.enable();
    map.scrollZoom.enable();
  },
});
</script>

<style lang="scss" scoped>
.modal {
  position: fixed;
  top: 50%;
  left: 50%;
  max-width: 560px;
  width: calc(100% - (56px + 2rem));
  padding: 32px 28px 16px;
  background: #fff;
  border-radius: 15px;
  box-shadow: 0 4px 16px rgba($color: #000000, $alpha: 0.1);
  transform: translate(-50%, -50%);
  z-index: 10;
}
</style>
