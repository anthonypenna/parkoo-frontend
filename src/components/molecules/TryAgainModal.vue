<template>
  <AlertModal @confirm="onTryAgain">
    <slot />
    <template #button>Try again</template>
  </AlertModal>
</template>

<script lang="ts">
import Vue from "vue";
import AlertModal from "@/components/molecules/AlertModal.vue";

export default Vue.extend({
  name: "TryAgainModal",
  components: { AlertModal },
  methods: {
    onTryAgain() {
      this.$emit("tryagain");
    },
  },
});
</script>
