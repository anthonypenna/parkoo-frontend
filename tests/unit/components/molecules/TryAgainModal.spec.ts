import { mount } from '@vue/test-utils'
import TryAgainModal from '@/components/molecules/TryAgainModal.vue'
import Button from '@/components/atoms/Button.vue'
import Modal from '@/components/atoms/Modal.vue'

describe('<TryAgainModal />', () => {
  it('should render correctly', () => {
    const wrapper = mount(TryAgainModal, {
      slots: { default: 'Hello world!' },
      stubs: { Modal: true },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('when clicking on the try again button', () => {
    it('should emit a try again event', () => {
      const wrapper = mount(TryAgainModal, { stubs: { Modal: true } })
      wrapper.findComponent(Button).trigger('click')
      expect(wrapper.emitted().tryagain).toBeTruthy()
    })
  })
})
