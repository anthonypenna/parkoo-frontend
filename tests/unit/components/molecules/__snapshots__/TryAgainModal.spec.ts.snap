// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TryAgainModal /> should render correctly 1`] = `<modal-stub>Hello world! <button class="button button--secondary">Try again</button></modal-stub>`;
